code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -24,76 +77,156 @@
+	playsound(owner, "sound/effects/xeno_newlarva.ogg", 25, 0, 1)
 	return ..()
 
-
 /datum/action/xeno_action/stealth/action_activate()
-	var/mob/living/carbon/xenomorph/hunter/stealthy_beno = owner
-	if(stealthy_beno.stealth)
-		stealthy_beno.cancel_stealth()
+	if(stealth)
+		cancel_stealth()
 		add_cooldown()
 		return TRUE
 
 	succeed_activate()
-	to_chat(stealthy_beno, "<span class='xenodanger'>We vanish into the shadows...</span>")
-	stealthy_beno.last_stealth = world.time
-	stealthy_beno.stealth = TRUE
-	stealthy_beno.RegisterSignal(stealthy_beno, COMSIG_MOVABLE_MOVED, /mob/living/carbon/xenomorph/hunter/.proc/handle_stealth_movement)
-	stealthy_beno.handle_stealth()
+	to_chat(owner, "<span class='xenodanger'>We vanish into the shadows...</span>")
+	last_stealth = world.time
+	stealth = TRUE
+	RegisterSignal(owner, COMSIG_MOVABLE_MOVED, .proc/handle_stealth)
+	handle_stealth()
 	add_cooldown()
-	addtimer(CALLBACK(stealthy_beno, /mob/living/carbon/xenomorph/hunter/.proc/sneak_attack_cooldown), HUNTER_POUNCE_SNEAKATTACK_DELAY) //Short delay before we can sneak attack.
-
+	addtimer(CALLBACK(src, .proc/sneak_attack_cooldown), HUNTER_POUNCE_SNEAKATTACK_DELAY) //Short delay before we can sneak attack.
 
-/mob/living/carbon/xenomorph/hunter/proc/cancel_stealth() //This happens if we take damage, attack, pounce, toggle stealth off, and do other such exciting stealth breaking activities.
+/datum/action/xeno_action/stealth/proc/cancel_stealth() //This happens if we take damage, attack, pounce, toggle stealth off, and do other such exciting stealth breaking activities.
 	if(!stealth)//sanity check/safeguard
 		return
-	to_chat(src, "<span class='xenodanger'>We emerge from the shadows.</span>")
-	UnregisterSignal(src, COMSIG_MOVABLE_MOVED) //This should be handled on the ability datum or a component.
+	to_chat(owner, "<span class='xenodanger'>We emerge from the shadows.</span>")
+	UnregisterSignal(owner, COMSIG_MOVABLE_MOVED) //This should be handled on the ability datum or a component.
 	stealth = FALSE
 	can_sneak_attack = FALSE
-	alpha = 255 //no transparency/translucency
+	owner.alpha = 255 //no transparency/translucency
 
-// ***************************************
-// *********** Pounce/sneak attack
-// ***************************************
-/datum/action/xeno_action/activable/pounce/hunter
-	plasma_cost = 20
-	range = 7
+/datum/action/xeno_action/stealth/proc/sneak_attack_cooldown()
+	if(!stealth || can_sneak_attack)
+		return
+	can_sneak_attack = TRUE
+	to_chat(owner, "<span class='xenodanger'>We're ready to use Sneak Attack while stealthed.</span>")
+	playsound(owner, "sound/effects/xeno_newlarva.ogg", 25, 0, 1)
 
-/datum/action/xeno_action/activable/pounce/hunter/prepare_to_pounce()
-	. = ..()
+/datum/action/xeno_action/stealth/proc/handle_stealth()
+	if(!stealth)
+		return
+	var/mob/living/livingowner = owner
+	if(owner.stat != CONSCIOUS || owner.lying_angle || livingowner.resting) //Can't stealth while unconscious/resting

```

Comment:
Use stuff like this to substitute these checks:
`RegisterSignal(owner, list(SIGNAL_ADDTRAIT(TRAIT_KNOCKEDOUT), SIGNAL_ADDTRAIT(TRAIT_FLOORED)), .proc/cancel_stealth)`

Reply:
is that merged